app/src/main/java/org/mozilla/focus/utils/UrlUtils.java

```diff
@@ -123,7 +123,10 @@ public static boolean isInternalErrorURL(final String url) {
         return "data:text/html;charset=utf-8;base64,".equals(url);
     }
 
-    public static boolean urlsMatchExceptForTrailingSlash(final @NonNull String url1, final @NonNull String url2) {
+    public static boolean urlsMatchExceptForTrailingSlash(final String url1, final String url2) {
```

Comment:
**Analysis**:
Assuming `request.getUrl().toString()` is not null (and assuming the impl of abstract Uri is OpaqueUri, it's not: http://androidxref.com/8.0.0_r4/xref/frameworks/base/core/java/android/net/Uri.java#961), `currentPageURL` must be null. It's assigned in two places which I analyzed:
- TrackingProtectionWebViewClient.notifyCurrentURL: all of the callers null check this value before setting it
- TPWVC.onPageStarted: this value is *not* null checked and the value is passed from the webView APIs (maybe it crashes on only some API levels?)

So I think the null is getting set from the latter.

---

It looks like the code won't crash (it seems like currentURL basically gets ignored if it's null), but I think we should fix the assumption that currentURL should not be null: can you file a bug to look into this further and fix it? (and CC Sebastian)